android/app/src/main/java/org/fossasia/openevent/utils/ISO8601Date.java

```diff
@@ -51,12 +51,12 @@ public static String dateFromCalendar(Calendar currentDate) {
     }
 
     public static String getTimeZoneDateFromString(String dateString) throws ParseException {
-        SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-DD", Locale.getDefault());
+        SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd", Locale.getDefault());
         formatter.setTimeZone(getEventTimezone());
         Date date = formatter.parse(dateString);
-        SimpleDateFormat newFormat = new SimpleDateFormat("d MMM", Locale.getDefault());
-
-        return newFormat.format(date);
+        SimpleDateFormat newFormat = new SimpleDateFormat("d MMM, z", Locale.getDefault());
```

Comment:
Ok. For an event based application, these date errors are crucial. I would have recommended Joda Time library but it's size is huge relative to the convenience it provides. What we can do is write tests. Currently, I am busy with exams so probably you or some other person can take up the issue to create unit tests for date conversion code?